src/main/java/com/es/phoneshop/model/cart/CartService.java

```diff
@@ -2,7 +2,9 @@
 
 import com.es.phoneshop.model.exceptions.OutOfStockException;
 
+import javax.servlet.http.HttpSession;
+
 public interface CartService {
-    Cart getCart();
-    void add(Long productId, int quantity) throws OutOfStockException;
+    Cart getCart(HttpSession httpSession);
```

Comment:
I believe this solution is the best option to divide business logic in service from specific technology.  HttpSession -- it is not a Servlet API technology, it is the way a browser communicates with a server. Nevertheless, i asked Andrei also what did he mean)